superduperdb/backends/mongodb/data_backend.py

```diff
@@ -217,3 +217,21 @@ def create_table_and_schema(self, identifier: str, mapping: dict):
         :param identifier: The identifier for the table
         :param mapping: The mapping for the schema
         """
+
+    def auto_create_table_schema(self, db, table_name, documents):
+        """Auto create table schema.
+
+        For MongoDB, this will infer the schema and apply it to the documents.
+        We use inlined schema for MongoDB because it is schema-less.
+
+        :param db: The datalayer instanace
+        :param table_name: The table name
+        :param documents: The documents
+        """
+        schema_dict = {}
+        for document in documents:
+            schema = self.infer_schema(document)
+            if schema.fields:
+                schema_dict[schema.identifier] = schema
+                document.schema = schema
+            db.apply(list(schema_dict.values()))
```

Comment:
Why not `db.apply(schema)`?

Reply:
The indentation here is incorrect; it should be added uniformly at the end. For data of the same format, the schemas they infer should be identical. After deduplication using the identifier, they should be added all at once.

I will fix this.

Reply:
Ok, but I think that we could potentially gain flexibility like this. At the program start, if `CFG.auto_schema == True`, we could load all datatypes, and use those with `check` set to check data.

Let's think about this later if developers need this feature.

Reply:
OK